### ISSUE TITLE
Tried to run diffusers/stable_diffusion.ipynb on Mac M3 and ran into errors

### ISSUE THREAD

**Issue by chrismattmann** [huggingface/diffusers#9967]
### Describe the bug

I tried to run diffusers/stable_diffusion.ipynb on Mac M3 and it failed in several parts, including:

* nvidia-smi call doesn't work
* wasn't able to download the torch models due to some weird issues like `ImportError: cannot import name 'DIFFUSERS_SLOW_IMPORT' from 'diffusers.utils'` due to conflicting libraries
* cuda calls needed to be replaced with `mps` device

I went ahead and updated the notebook and have a PR I could submit if interested that fixes it. I just made a copy of the notebook and added _macosx.ipynb to it. I realize some fancier way could have been to try some if/else guess system types in the notebook else, but I just opted for fixing it separately.

### Reproduction

Tried on a brand new Mac M3 

### Logs

```shell
Don't really have logs
```

### System Info

Brand new Mac M3 Max running Sonoma 14.6

**Comment by chrismattmann** [comment]
Very excited b/c I fixed it! It works locally perfectly now! Love the image generation and thank you for your work. Let me know if you'd like the PR!

**Comment by sayakpaul** [comment]
Of course, please feel free PR it :) Thank you! Cc: @pcuenca

**Comment by chrismattmann** [comment]
excellent @sayakpaul i will prepare and submit today!

**Comment by chrismattmann** [comment]
Added PR in https://github.com/huggingface/notebooks/pull/533

**Comment by chrismattmann** [comment]
@sayakpaul @pcuenca

**Comment by pcuenca** [comment]
> wasn't able to download the torch models due to some weird issues

How did you fix this?

**Comment by chrismattmann** [comment]
> > wasn't able to download the torch models due to some weird issues
> 
> How did you fix this?

@pcuenca the fix was this line:
 ```
   if isosx:
    # This is added to get around some issues of Torch not loading models correctly on Mac OS X
    !pip install --upgrade huggingface-hub==0.26.2 transformers==4.46.1 tokenizers==0.20.1 diffusers==0.31.0
 ```

I Googled around and found someone saying that you need these versions of those libraries to make it work on Mac OSX and this worked for me.

**Comment by chrismattmann** [comment]
fixed in #533
